### DISCUSSION TITLE
Importing Visio stencils from visiocafe.com

### DISCUSSION THREAD

**Discussion by tfboy** [jgraph/drawio-desktop#1405] [Q&A]
I use Wiki.js for documenting at my workplace and have a long history of old Visio documents I'd like to bring into drawio.

However, it appears that any drawing including shapes provided by the Visio stencils from visiocafe.com don't appear to work. I just get a generic recrangle with a red border and line through it.

My guess is because these shapes have additional data in them that is incomprehensible to drawio.

By my dream would be to be able to bring the drawings and reuse the stencils in drawio.

Has anyone managed to do this or can offer any tips to make the shapes usable in drawio?

**Comment by davidjgraph** [comment]
We'd need exact steps and re-production case.

**Reply by tfboy** [reply]
Download (for example), an HP stencil from Visiocafe: http://www.visiocafe.com/downloads/hp/HPE-ProLiant.zip
Open up a .vss file in Visio
Save the .vss file to a .vssx stencil format
In Wiki.js, go to edit a page and then click to open the draw.io editor.
In the draw.io editor, go to file -> import -> from peripheral.
Select the .vssx file.
It does then appear in the left, but none of the models are are correctly shown; all have a red rectangle and line through them:
![image](https://github.com/jgraph/drawio-desktop/assets/6972726/4154eef2-278c-4f8f-95c2-980cc8ef265d)

**Reply by davidjgraph** [reply]
If you do it in app.diagrams.net, does it work?

**Comment by tfboy** [comment]
Also, without going to the trouble of importing the vssx stencil in the embedded drawio, you can just drag one of the shapes into Visio, save the Visio drawing as a .vsdx and then open that in the desktop app of drawi.io and the issue is the same:
![image](https://github.com/jgraph/drawio-desktop/assets/6972726/1fd00a82-c78d-45fe-a92a-7343bf1951ad)

**Reply by davidjgraph** [reply]
That won't work, these are EMF images, they require a server round-trip.

**Reply by tfboy** [reply]
Thanks @davidjgraph - was your comment regarding EMF images for the desktop version specifically, or for all versions?

If I try the online version app.diagrams.net, interestingly, some of them work, some don't.

![image](https://github.com/jgraph/drawio-desktop/assets/6972726/22e12de5-fe9f-4660-86b7-0811c37db436)

**Reply by davidjgraph** [reply]
Specifically with desktop, it cannot perform vss import or import with EMFs embedded.

It's a pretty large set and we're hitting timeouts on that. I think we can improve on this so that the VC sets import, at the least. Give us a few weeks.

**Reply by tfboy** [reply]
Thanks.
If I knew Visio better, I'd have a go at splitting them into smaller stencil sets. Most of those I don't use anyway. I'll have a look if I can delete the unused ones to make the thing smaller.

**Reply by davidjgraph** [reply]
If you have a recent copy of MSV, start by loading the vss set in and then re-save in the newer vssx and import that to draw. That will have a better chance.

**Reply by tfboy** [reply]
That's what I've done so far. But I did try on two different PCs with different versions of Visio and I think the more recent version gave the latest screenshot with some devices working whereas the older version was a complete failure.

**Comment by tfboy** [comment]
Just to update this, I've now tried creating a new massively reduced stencil in MSV with only a couple of required server blocks and saving this as an vssx file. My resulting file is smaller, under 500kB. I'm then able to copy that to my client which has access to the bunkered Wiki instance.

When I try to import this, I don't get anything at all - just the whizzing wheel and Loading...
If I look at the network traffic, I believe it's trying to contact some remote sites like js-content-cache-1.ps5.canonical.com.

As my instance is bunkered with no Internet access, is there a way to ensure that any vssx files brought in have all the data embedded and not require downloading extra data from the Internet?

**Comment by tmanok** [comment]
Hi Everyone,

Over the last few months of starting a new job in a project department, I've had to do make diagrams for a significant number of server rooms with a with variety of computer equipment. Being as I prefer to work with DrawIO over Visio by a significant margin, I'm frequently testing new Visio stencils in Drawio. Unfortunately, most of them do not work and on occasion I am in need of something more niche than what is available. While there are fabulous online sources for logos and icons that work perfectly in Drawio, I am stumped in these rack diagrams. 

Just today on the desktop app (preferred due to offline capabilities) I attempted to load a .vssx from VisioCafe (https://www.visiocafe.com/lenovo.htm) Lenovo ThinkAgile HX specifically, and observed the following:
![Screenshot from 2024-05-07 16-30-31](https://github.com/jgraph/drawio-desktop/assets/29475831/35566408-1e66-4169-a074-176b9a1dd40e)

Those are green lines that are then previewed and placed as follows:
![Screenshot from 2024-05-07 16-31-19](https://github.com/jgraph/drawio-desktop/assets/29475831/4ecf34f0-b7d5-4ec5-bf5a-105f0fc0eb23)
![Screenshot from 2024-05-07 16-30-56](https://github.com/jgraph/drawio-desktop/assets/29475831/67068605-90f8-4ea5-bc51-498e33cd9836)

Meanwhile using app.diagrams.net the following similar behaviour occurs:
![Screenshot from 2024-05-07 16-37-54](https://github.com/jgraph/drawio-desktop/assets/29475831/57461d33-644e-49ea-bbb7-bf7dc3948dd2)
![Screenshot from 2024-05-07 16-37-24](https://github.com/jgraph/drawio-desktop/assets/29475831/d80db1c1-30b1-4281-b20d-bcd13a12607e)

Thank you!


Tmanok

**Comment by bitfim** [comment]
I also would like to use shapes and stancils from visiocafe, but its not possible to use .vss files directly with draw.io web of windows-app.

Also, saving shapes and stancils into a .vsdx file and importing this to draw.io web or desktop does not work. 

Would be nice to get this fixed asap.

A workaround is to copy the shapes and stancils in visio and paste it into draw.io web or desktop. But this only imports a picture and not a vector graphic.

**Reply by davidjgraph** [reply]
Would be nice if I won the lottery asap, but...y'know...

**Reply by bitfim** [reply]
Sorry, I didn't mean to push.

I tested a bit more and noticed, that its possible to drag and drop .vss files into the web version of draw.io, but not into the desktop version (windows). But not all stanciles are intact. And I dont mean the same behavior like on @tmanok s pictures. Its more like that some stancils are devided into its single layers and functions.

**Reply by davidjgraph** [reply]
Precisely which file did you test?

**Reply by bitfim** [reply]
I followed this guide: https://www.drawio.com/doc/faq/open-vsd-files-confluence-server

I downloaded the [Dell-Networking.zip](http://www.visiocafe.com/downloads/dell/Dell-Networking.zip) dated to 17-Oct-2024,
opened the Dell-Networking-FixedPorts.vss from the zip-file with MS Visio,
created a new diagram in Visio,
added the following devices to the the newly created diagrad: S5212-ON, S5248F-ON and S5232F-ON,
and saved the .vsdx-file to my local drive. 

This is how the .vsdx-file looks in Visio: 

![grafik](https://github.com/user-attachments/assets/d83f8fbd-4633-48d7-b7d0-e173948818a5)
Note: I also added a Arista device for reference.

I then opened https://app.diagrams.net/ with Firefox or Edge or Chrome, 
created a new diagram,
drag and dropped the .vsdx-file into the new diagram and waited until all switches had been imported. 

The raw result looks like this:

![grafik](https://github.com/user-attachments/assets/7896aade-e5f2-453c-b56b-51a8b2f7c315)

The arista shape seems to be intact, but the Dell devices not. It looks like its devided into its single layers and I think that transparent colors are converted into white color:

![grafik](https://github.com/user-attachments/assets/5caf9899-bf2d-437f-8116-84b38bea3dcd)

**Reply by KittyKatt** [reply]
Can confirm the above is happening for me as well with the VisioCafe Dell collections on app.diagrams.net. Having a hard time finding another Dell shape library, unfortunately.

**Reply by davidjgraph** [reply]
@KittyKatt please open a new issue with a link to the set in question.

**Comment by archermask** [comment]
how to make thats stencils persistent after import? because in my drawio desktop they're not persistent. if I create new diagram, i must to import again what stencils i needed. 
thankyou
